### ISSUE TITLE
CaptchaRequired

### ISSUE THREAD

**Issue by matlink** [yeriomin/token-dispenser#23]
Hello,
Google asks me for Captcha using token dispenser like this
```
com.github.yeriomin.playstoreapi.AuthException: Auth error body: Error=CaptchaRequired
```
How can it be handled?
Thank you.

**Comment by archon810** [comment]
Getting the same all the time now.

**Comment by archon810** [comment]
Is it possible Google is looking at the user agent as well now, and the dispenser doesn't seem to be setting any? @yeriomin Any ideas?

**Comment by matlink** [comment]
Some friends using token dispenser have not this issue even if they use the latest version.

**Comment by archon810** [comment]
I've tried 5 different servers and 2 different accounts and they all have the captcha challenge. It used to only pop up if you haven't set up 2FA and app specific passwords, with a link to solve the Captcha, but now it's happening to app specific passwords too, and there's no link to solve the Captcha anymore.

**Comment by felipe-d-domingues** [comment]
Having the same problem here, but I'm getting a link to solve the Captcha
`2019-11-06 09:59:39:569 -0200 WARN com.github.yeriomin.tokendispenser.Server - com.github.yeriomin.playstoreapi.AuthException: Auth error body: Error=CaptchaRequired
CaptchaToken=xxxxxxxxxxxxxxxxxxxxxx
CaptchaUrl=https://accounts.google.com/Captcha?ctoken=xxxxxxxxxxxxxxxxxxxxxx&v=2
` 
(changed the actual token to xxxxxxxxxxxxxxxxxxxxxx for obvious reasons)
It used to work before, but now every single account is showing this error. How do you handle these links? No 2FA enabled

**Comment by archon810** [comment]
Right, the link just shows the Captcha image, that's all. Not sure how to proceed.

**Comment by archon810** [comment]
https://github.com/NoMore201/googleplay-api/issues/67

**Comment by tclavier** [comment]
An other option, deploy an application they upload apk from rooted phones ?

**Comment by hoffie** [comment]
I have analyzed this issue in another project which uses the same API (Android App Aurora Store). My results so far are that Google seems to decide whether to require a Captcha by analyzing TLS protocol details. More specifically, I was able to make it work by forcing specific ciphers and disabling ALPN.

I still have to figure out how to convince Java to set the proper TLS options (only did it successfully in Python so far).

I am detailing my results here: https://gitlab.com/AuroraOSS/AuroraStore/issues/217#note_249390026

**Comment by matlink** [comment]
@hoffie Very nice! I hope you can come back here and post results when you have found a clean solution to this! Thanks!

**Comment by hoffie** [comment]
Aurora Store seems to use a different login method now (/EmbeddedSetup with OAuth?), as does microg:
https://gitlab.com/AuroraOSS/AuroraStore/commit/1d7a82b01976041a83b5cbcef16ccd418e0b1225

Therefore, I'm currently no longer working on this, but happy to provide any relevant details from my tesst.

**Comment by halo2me** [comment]
@matlink I think you can refer to https://github.com/ClaudiuGeorgiu/PlaystoreDownloader

**Comment by jr3074** [comment]
hello, did you solve this?

**Comment by ildar** [comment]
No solution known yet. People tend to migrate to Aurora

**Comment by jr3074** [comment]
@ildar did you success to run auroradispenser?

**Comment by ildar** [comment]
Aurora doesn't use tokens anymore. See the gitlab link above for details.

**Comment by ouaguid** [comment]
Instead of using your account password (which can generate a validation request via captcha if 2-Step-Verification is enabled), it is preferable to generate and use an App password specific to your application by following the steps mentioned here => [Sign in with App Passwords] (https://support.google.com/accounts/answer/185833?hl=en)  (see the section "Create & use App Passwords").
